src/sqlfluff/core/rules/std/L044.py

```diff
@@ -104,7 +104,7 @@ def _analyze_result_columns(
                             if wildcard_table in queries:
                                 # Wildcard refers to a CTE. Analyze it.
                                 self._analyze_result_columns(
-                                    queries[wildcard_table], dialect, queries
```

Comment:
This fixes another issue I discovered in tracking down the issue: A CTE that queries itself was also causing infinite recursion.

Reply:
NIce 👍 

Initially I didn't get this, but I think it's actually quite an elegant solution.